addons/purchase/migrations/7.0.1.1/post-migration.py

```diff
@@ -49,36 +53,22 @@ def migrate_purchase_order_line_names(cr, pool):
 
 
 def set_purchase_order_payment_term(cr, pool):
+    """Set a payment_term_id for each purchase_order, using the payment term
+    associated to the purchase order supplier (that it's a property field).
     """
-    Set a journal_id for each purchase_order, using the script similar to
-    the function _get_journal_id, used in V7 when creating a new purchase
-    order.
-    """
-    partner_obj = pool.get('res.partner')
-    purchase_order_obj = pool.get('purchase.order')
-    res_user_obj = pool.get('res.users')
-    cr.execute(
-        "SELECT id, partner_id, create_uid, write_uid FROM purchase_order")
-
-    for (purchase_order_id, partner_id, create_uid, write_uid) in \
-            cr.fetchall():
-        # get the property as viewed by the partner who created / modified
-        # the purchase order.
-        if write_uid:
-            uid = write_uid
-        else:
-            uid = create_uid
-        my_user = res_user_obj.browse(cr, SUPERUSER_ID, uid)
-        partner_company_id = partner_obj.browse(
-            cr, SUPERUSER_ID, partner_id).company_id.id
-        if my_user.company_id.id != partner_company_id:
-            res_user_obj.write(
-                cr, SUPERUSER_ID, uid, {'company_id': partner_company_id})
-        supplier = partner_obj.browse(cr, uid, partner_id)
-        purchase_order_obj.write(
-            cr, SUPERUSER_ID, [purchase_order_id],
-            {'payment_term_id':
-             supplier.property_supplier_payment_term.id})
+    _logger.info("Setting payment term on purchase orders...")
+    openupgrade.logged_query(
+        cr,
+        """
+        UPDATE purchase_order
+        SET payment_term_id =
+            substring(ir_property.value_reference from 22)::integer
+        FROM ir_property
+        WHERE ir_property.name = 'property_supplier_payment_term'
+        AND ir_property.company_id = purchase_order.company_id
+        AND ir_property.res_id =
+            'res.partner,' || purchase_order.partner_id::text
```

Comment:
this is similar as above, also support global payment terms with `ir_property.res_id is null`